### ISSUE TITLE
S3M playback error

### ISSUE THREAD

**Issue by absolutelynothinghere** [DeaDBeeF-Player/deadbeef#2463]
### Steps to reproduce the problem

Try playing/queuing any of the following tracker module files:

* https://u.teknik.io/6rZmM.S3M
* https://u.teknik.io/s8DK8.S3M
* https://u.teknik.io/nbJ4J.S3M
* https://u.teknik.io/9CfBi.S3M

### What's going on? Describe the problem in as much detail as possible.

Trying to play/enqueue any of the files linked above produces errors and refuses to add them to playlist. Note that these files play fine in foobar2000 and audacious.

```
ERROR: could not load: /home/bob/Music/DANCINGS.S3M
ERROR: could not load: /home/bob/Music/FLAIR.S3M
ERROR: could not load: /home/bob/Music/FOR-YOU.S3M
ERROR: could not load: /home/bob/Music/HUNT.S3M
```

### Information about the software:

Deadbeef version: 1.8.3
OS: Void Linux

**Comment by Alexey-Yakovenko** [comment]
Can you verify whether foobar2000 can load these files using foo_dumb?
https://wiki.hydrogenaud.io/index.php?title=Foobar2000:Components/DUMB_Module_Decoder_(foo_dumb)

**Comment by Alexey-Yakovenko** [comment]
@kode54 I found your name in the code of reads3m.c, so would be great if you can provide some more educated opinion whether this change makes any sense :) Also wondering if newer version of foo_dumb already can handle those files.. Perhaps I should update?

**Comment by Alexey-Yakovenko** [comment]
More info: the last pattern has length=0, and 2 last samples are cut off (looks like EOF is reached before they're fully read), so they will get 0-padded instead of just failing. I don't like the idea of supporting corrupt files, but also I'm not entirely sure if they're corrupt or just use some weird variation of the format.

**Comment by Alexey-Yakovenko** [comment]
I compared with this file https://git.lopez-snowhill.net/chris/dumb/-/blob/master/src/it/loads3m.c
and they seem the same.

**Comment by Alexey-Yakovenko** [comment]
Ugh. Coincidentally, I feel D.U.M.B now. compared wrong files :)
I now compared the reads3m.c instead, and there seem to be some differences.

**Comment by Alexey-Yakovenko** [comment]
Replacing with your reads3m.c doesn't solve the issue, the files are not recognized. So for now I'll assume they won't play in foo_dumb either.

**Comment by kode54** [comment]
I don't even maintain DUMB any more, although if there are serious bugs, I will try to fix them eventually. I wonder if libopenmpt decodes the files in question? It also has a slightly less accurate, but way faster, length and subsong probe process.

**Comment by Alexey-Yakovenko** [comment]
Ok, thanks. I didn't point this out clearly enough - but I did make a fix for this situation: https://github.com/DeaDBeeF-Player/deadbeef/commit/b80c0cf8126352c8c06684e0db091eb88d3905d0
(In case you're interested).

I should check out libopenmt too at some point :)

**Comment by absolutelynothinghere** [comment]
> I did make a fix for this situation: [b80c0cf](https://github.com/DeaDBeeF-Player/deadbeef/commit/b80c0cf8126352c8c06684e0db091eb88d3905d0)

Sorry just saw this, thanks for fixing! 😀

> I should check out libopenmpt too at some point :)

Audacious uses libopenmpt and it plays my files correctly, so I'm sure DB will benefit greatly from it.
